packages/react-storage/src/components/StorageBrowser/views/Controls/Checkbox.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+
+import { CLASS_BASE } from '../constants';
+import {
+  ViewElement,
+  LabelElement,
+  InputElement,
+} from '../../context/elements/definitions';
+
+const BLOCK_NAME = `${CLASS_BASE}__checkbox`;
+const LABEL_CLASSNAME = `${BLOCK_NAME}-label`;
+
+interface CheckboxControlProps {
+  checked: boolean;
+  disabled?: boolean;
+  id: string;
+  labelHidden?: boolean;
+  labelText: string;
+  onSelect?: () => void;
+}
+
+export const CheckboxControl = ({
+  disabled,
+  checked,
+  onSelect,
+  labelText,
+  labelHidden = false,
+  id,
+}: CheckboxControlProps): React.JSX.Element => (
+  <ViewElement className={BLOCK_NAME}>
+    <InputElement
+      checked={checked}
+      className={`${BLOCK_NAME}-input`}
+      disabled={disabled}
+      id={id}
+      onChange={onSelect}
+      type="checkbox"
+    />
+    <LabelElement
+      className={[
+        LABEL_CLASSNAME,
+        labelHidden ? `${CLASS_BASE}-visually-hidden` : undefined,
```

Comment:
Think we can drop the visually hidden `className` by adding: 
```
style={labelHidden ? { display: 'none' } : undefined}
```

Reply:
`display: none` will make the text unreadable by screen readers. The visually hidden class will make it invisible to sighted users but accessible to screen readers.